out_writeapi.go

```diff
@@ -145,27 +156,69 @@ func checkResponses(curr_ctx context.Context, currQueuePointer *[]*managedwriter
 				recvOffset, err := queueHead.GetResult(curr_ctx)
 				*currQueuePointer = (*currQueuePointer)[1:]
 				if err != nil {
-					log.Fatal("error in checking responses")
-					return 1
+					log.Printf("Append response error: %s", err)
+					return 1, err
 				}
 				log.Printf("Successfully appended data at offset %d.\n", recvOffset)
 			default:
-				return 0
+				return 0, nil
 			}
 		}
 
 	}
-	return 0
+	return 0, nil
+}
+
+func getConfigField(plugin unsafe.Pointer, key string, errmsg string, defaultval int) int {
+	currstr := output.FLBPluginConfigKey(plugin, key)
+	var finval int
+	if currstr == "" {
+		finval = defaultval
+	} else {
+		finval, err = strconv.Atoi(currstr)
+		if err != nil {
+			log.Printf("%s: %s", errmsg, err)
+			finval = defaultval
+		}
+	}
+	return finval
+}
+
+type ManagedWriterClient interface {
```

Comment:
Could you add a comment above this interface that explains why it is needed?

Also, I suggest we simply clone the entire existing ManagedWriter interface, even if we are not currently using all of it. That way it will be easy to begin using the other functions when needed.